src/algebra/pointwise.lean

```diff
@@ -410,6 +410,28 @@ by { convert finset.card_image_le, rw [finset.card_product, mul_comm] }
 lemma mul_card_le [has_mul α] {s t : finset α} : (s * t).card ≤ s.card * t.card :=
 by { convert finset.card_image_le, rw [finset.card_product, mul_comm] }
 
+open_locale classical
+
+/-- A finite set `U` contained in the product of two sets `S * S'` is also contained in the product
+of two finite sets `T * T' ⊆ S * S'`. -/
+@[to_additive]
+lemma subset_mul {M : Type*} [monoid M] {S : set M} {S' : set M} {U : finset M} (f : ↑U ⊆ S * S') :
+  ∃ (T T' : finset M), ↑T ⊆ S ∧ ↑T' ⊆ S' ∧ U ⊆ T * T' :=
+begin
+  apply finset.induction_on' U,
+  { use [∅, ∅], simp only [finset.empty_subset, finset.coe_empty, set.empty_subset, and_self], },
+  rintros a s haU hs has ⟨T, T', hS, hS', h⟩,
+  obtain ⟨x, y, hx, hy, ha⟩ := set.mem_mul.1 (f haU),
+  use [insert x T, insert y T'], simp only [finset.coe_insert],
+  repeat {rw [set.insert_subset], }, use [hx, hS, hy, hS'],
+  refine finset.insert_subset.mpr _,
+  split, rw finset.mem_mul, use [x,y],
```

Comment:
This `split` creates two goals. Please put `{ }` around the goals to indicate this.

Reply:
I am forgetful about that, thank you for pointing it out! I have made the changes.